src/software/thunderscope/replay/replay_corruption_test.py

```diff
@@ -0,0 +1,189 @@
+    invalid log  entry are created
+
+
+    :list_of_protos: the list of proto that we are referencing when creating the log entries
+    :save_path: where we are saving the replay file
+    :duration: how long do we want t create this replay chunks 
+    :start_time: when is the replay chunk being started? 
+    :gen_log_entry_func: the function that is used to generate invalid log entries 
+    :frequency: what percent of the time should we call gen_log_entry_func
+    """
+    os.makedirs(save_path, exist_ok=True)
+    path_to_repaly_file = os.path.join(save_path, "0.replay")
+
+    with gzip.open(path_to_repaly_file, "ab") as log_file:
+        for i in range(len(list_of_protos)): 
+            # calculate time based on linear interpolation
+            proto_time = start_time + (duration - 0)/len(list_of_protos) * i   
+            proto = list_of_protos[i]
+
+            if random.random() > frequeny: 
+                # this would happen 90 % of the time
+                log_entry = create_valid_log_entry(proto, proto_time)
+            else: 
+                # this would happen 10% of the time
+                log_entry = gen_log_entry_func(proto, proto_time)
+
+            ProtoLogger.write_to_logfile(log_file, bytes(log_entry, encoding="utf-8"))
+
+def make_replay_chunk(size_of_replay_chunk=1000):
+    """
+    Creating a replay chunks that is like the following: 
+    1. from 0 to 0.1 seconds, it all of the entries are valid
+    2. from 0.1 to 0.2 seconds, 10% of the proto cannot be decoded 
+    3. from 0.2 to 0.3 seconds, 10% of the proto are missing delimeter
+
+    :size_of_replay_chunks: the size of the replay chunks we want to make
+    """
+    # making sure previous test results doesn't affect the new test
+    shutil.rmtree(TMP_REPLAY_SAVE_PATH, ignore_errors=True)
+
+    replay_proto =  []
+    # we divide by three because we make three copies in the code below, so we have
+    # close to size_of_replay_chunk in the end, assuming integer rounding that is. 
+    for _ in range(int(size_of_replay_chunk/3)):
+        some_proto = create_random_proto()
+        replay_proto.append(some_proto)
+
+    # making the replay chunks 
+    # the first part of the replay chunks is from  0 to 0.1 seconds, it all of the entries are valid
+    # the seconds part of the replay chunk is from 0.1 to 0.2 seconds, 10% of the proto cannot be decoded 
+    # the third part of the replay chunk is from 0.2 to 0.3 seconds, 10% of the proto are missing delimeter
+    make_part_replay_chunks(replay_proto, TMP_REPLAY_SAVE_PATH, 0.1, 0.0, create_valid_log_entry)
+    make_part_replay_chunks(replay_proto, TMP_REPLAY_SAVE_PATH, 0.1, 0.1, create_corrupt_log_entry)
+    make_part_replay_chunks(replay_proto, TMP_REPLAY_SAVE_PATH, 0.1, 0.2, create_missing_delimeter_log_entry)
+
+def cleanup():
+    """
+    cleaning up this test by deleting the replay files 
+    """
+    shutil.rmtree(TMP_REPLAY_SAVE_PATH)
+
+def create_test_player()() -> ProtoPlayer:
```

Comment:
CI seems to be failing here
```suggestion
def create_test_player() -> ProtoPlayer:
```

Reply:
That was an low key error on my part. I forgot to push the latest branch.